src/Risk.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Risk where
+
+import Control.Monad
+import Data.List
+import Control.Monad.Random
+
+------------------------------------------------------------
+-- Die values
+
+newtype DieValue = DV { unDV :: Int }
+  deriving (Eq, Ord, Show, Num)
+
+first :: (a -> b) -> (a, c) -> (b, c)
+first f (a, c) = (f a, c)
```

Comment:
You don't need to implement `first` function manually, just import `first` from `Data.Bifunctor` module